src/Directory.Build.props

```diff
@@ -36,7 +36,7 @@
   </PropertyGroup>
 
   <!-- Only compile with IsAotCompatible on .NET 8.0 and later -->
-  <PropertyGroup Condition="$([System.Text.RegularExpressions.Regex]::IsMatch('$(TargetFramework)', 'net8\.|net9\.|net\d{2,}\.'))">
+  <PropertyGroup Condition="$([System.Text.RegularExpressions.Regex]::IsMatch('$(TargetFramework)', '(net8\.0|net9\.0|net\d{2,}\.0).*'))">
```

Comment:
This is hard to read and error prone, would it be too noisy to add the actual TFMs stack line by line?

Reply:
or just net8, like in rest of the code

Reply:
But yeah, I guess the fact that I'm correcting a broken RegEx in this PR is not a good sign...